### PR TITLE
Implement Add, Edit and Remove task functionalities using ES6 syntax.

### PR DESCRIPTION
**Pull request by Mo7amedSMahdi**
- Create modules folder.
- Create task class in index.js.
- Create form.js to get the user tasks.
- Create functionalities.js to implement Add, Edit, Remove task functionalities.
- Save and retrive tasks from local storage.

### CODE REVIEW

File: src/modules/functionalities.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+      } else {
+        removeTask(tasks, addTask, item);
+      }
+
+      newDesc = '';
+      deletedTask = false;
+    });
+  });
+};
```
Comment (Whoistolu) [inline src/modules/functionalities.js:84]: - [x] [REQUIRED] Kindly ensure that a new task can be added, edited, and deleted. Currently, I cannot add any tasks as the error states: `Cannot Post`.
- Kindly ensure that by default, new tasks should have the property `completed` set to `false` and the property `index` set to the value of the new array length.
- Also, ensure that deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- Please make sure that all changes to the To-Do List should be saved in local storage.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @Mo7amedSMahdi 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/functionalities.js (hunk 1 above)
Comment (Mo7amedSMahdi) [reply src/modules/functionalities.js:84]: Thanks a lot :pray: , for your understanding now I fixed :muscle: the problem and everything is supposed to run as expected.

**Review by keroloussamy — APPROVED** [review]
Hi @Mo7amedSMahdi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap

## Optional suggestions
_Check the comments under the review._

- [x] x

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**